src/fmu/dataio/_model/root.py

```diff
@@ -210,6 +210,7 @@ class FmuResultsJsonSchema(GenerateJsonSchema):
         "data.is_prediction",
         "data.name",
         "data.offset",
+        "data.product",
```

Comment:
Perhaps be more granular on sub-fields under `data.product` rather than making the whole thing "contractual". That way, some subfields can be assumed stable (indexed in Sumo, used by consumers, etc) while others might still be added/removed/changed.